src/main/java/catan/services/impl/PlayServiceImpl.java

```diff
@@ -125,17 +131,26 @@ private void endTurn(GameBean game) throws GameException {
             Integer newPreparationCycle = game.getPreparationCycle();
             boolean preparationFinished = newPreparationCycle == null && !game.getCurrentMove().equals(1);
             shouldUpdateNextMove = (preparationFinished || previousPreparationCycle.equals(newPreparationCycle));
+        } else {
```

Comment:
don't see the ways how to refactor it fully now. how about switch-case instead of if-else? i think it would be a little bit more readable.